src/lib/shooting/DeviceOrientationControls.ts

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable */
+
+// https://github.com/mrdoob/three.js/issues/22996
+
+import { Euler, EventDispatcher, MathUtils, Quaternion, Vector3 } from 'three';
+
+const _zee = new Vector3(0, 0, 1);
+const _euler = new Euler();
+const _q0 = new Quaternion();
+const _q1 = new Quaternion(-Math.sqrt(0.5), 0, 0, Math.sqrt(0.5)); // - PI/2 around the x-axis
+
+const _changeEvent = { type: 'change' };
+
+class DeviceOrientationControls extends EventDispatcher {
+  object: any;
+  enabled: boolean;
+  deviceOrientation: any;
+  screenOrientation: number;
+  alphaOffset: number;
+  connect: () => void;
+  disconnect: () => void;
+  update: () => void;
+  dispose: () => void;
+
+  constructor(object: any) {
+    super();
+
+    if (window.isSecureContext === false) {
+      console.error(
+        'THREE.DeviceOrientationControls: DeviceOrientationEvent is only available in secure contexts (https)',
+      );
+    }
+
+    const scope = this;
+
+    const EPS = 0.000001;
+    const lastQuaternion = new Quaternion();
+
+    this.object = object;
+    this.object.rotation.reorder('YXZ');
+
+    this.enabled = true;
+
+    this.deviceOrientation = {};
+    this.screenOrientation = 0;
+
+    this.alphaOffset = 0; // radians
+
+    const onDeviceOrientationChangeEvent = function (event: any) {
+      scope.deviceOrientation = event;
+    };
+
+    const onScreenOrientationChangeEvent = function () {
+      scope.screenOrientation = window.orientation || 0;
```

Comment:
To resolve the deprecation warning for `window.orientation`, you can use the `screen.orientation` API instead. Here's a concise fix:

```typescript:src/lib/shooting/DeviceOrientationControls.ts
const onScreenOrientationChangeEvent = function () {
  scope.screenOrientation = screen.orientation ? screen.orientation.angle : 0;
};
```

This change uses the modern `screen.orientation.angle` property, which provides the same information as the deprecated `window.orientation`. The fallback to 0 is kept in case `screen.orientation` is not supported in some browsers.

```suggestion
      scope.screenOrientation = screen.orientation ? screen.orientation.angle : 0;
```